source/agora/utils/TxBuilder.d

```diff
@@ -319,20 +319,22 @@ public struct TxBuilder
         foreach (ref o; this.data.outputs)
             o.type = outputs_type;
 
+        // Sign all inputs using unlocker
+        foreach (idx, ref in_; this.inputs)
+            this.data.inputs[idx].unlock = unlocker(this.data, in_);
+
```

Comment:
That will not work, the signature needs to happen after we have all the `output` ready, otherwise it will be invalid.

Reply:
At first I thought this was only `inputs` being updated so could be done first, but you are right as the `unlocker` will sign the `tx` which needs all the outputs to be included.